respy/parallelization.py

```diff
@@ -68,6 +68,47 @@ def wrapper_parallelize_across_dense_dimensions(*args, **kwargs):
         return decorator_parallelize_across_dense_dimensions
 
 
+def combine_and_split_interpolation(func):
+    """Combine and split the information across sub state spaces.
+
+    For the interpolation, we compute endogenous, exogenous, and other components within
+    each sub state space. The linear model to predict the expected value function has to
+    be fitted on all states withing a period which is why this decorator combines the
+    information.
+
+    """
+
+    @functools.wraps(func)
+    def wrapper_combine_and_split_interpolation(
+        endogenous, exogenous, max_evf, not_interpolated
+    ):
+        if isinstance(endogenous, dict):
+            dense_indices = endogenous.keys()
+
+            endogenous = np.hstack(tuple(endogenous.values()))
+            exogenous = np.row_stack(tuple(exogenous.values()))
+            max_evf = np.hstack(tuple(max_evf.values()))
+            not_interpolated = np.hstack(tuple(not_interpolated.values()))
+
+            out = func(endogenous, exogenous, max_evf, not_interpolated)
+
+            period_expected_value_functions = {
```

Comment:
Can we be sure that the order is already preserved such that this will always work? The way I understand it now is that out is just an np.array?

Reply:
It is a numpy array. But the order of stacked rows does not change.